pytorch_lightning/loops/dataloader/evaluation_loop.py

```diff
@@ -271,17 +271,65 @@ def _on_evaluation_epoch_end(self) -> None:
+                if isinstance(v, dict):
+                    yield from self._get_keys(v)
+                else:
+                    yield k
+
+    def _find_value(self, data: dict, target: str) -> Iterable[Any]:
+        for k, v in data.items():
+            if k == target:
+                yield v
+            elif isinstance(v, dict):
+                yield from self._find_value(v, target)
+
     def _print_results(self, results: List[_OUT_DICT], stage: RunningStage) -> None:
-        # TODO: this could be updated to look nicer
-        from pprint import pprint
-
-        print("-" * 80)
-        for i, metrics_dict in enumerate(results):
-            print(f"DATALOADER:{i} {stage.upper()} RESULTS")
-            pprint(
-                {
-                    k: (v.item() if v.numel() == 1 else v.tolist()) if isinstance(v, torch.Tensor) else v
-                    for k, v in metrics_dict.items()
-                }
-            )
-            print("-" * 80)
+        # remove the dl idx suffix
+        results = [{k.split("/dataloader_idx_")[0]: v for k, v in result.items()} for result in results]
+        unique_keys = sorted(set(self._get_keys(results)))
+
+        if _RICH_AVAILABLE:
+            from rich.console import Console
+            from rich.table import Table
+
+            console = Console()
+            table = Table()
+
+            rows = [[key] for key in unique_keys]
+            table.add_column(f"{stage.capitalize()} Metric", justify="center", style="cyan")
+            for i, metrics in enumerate(results):
+                table.add_column(f"DataLoader {i}", justify="center", style="magenta")
+                for metric in rows:
+                    v = list(self._find_value(metrics, metric[0]))
+                    metric.append(f"{v[0]}" if v else " ")
+            for row in rows:
+                table.add_row(*row)
+
+            console.print(table)
+        else:
+            import os
+
+            rows = [[] for _ in unique_keys]
+            for i, metrics in enumerate(results):
+                for j in range(len(rows)):
+                    v = list(self._find_value(metrics, unique_keys[j]))
+                    rows[j].append(f"{v[0]}" if v else " ")
+
+            cols = [f"{stage.capitalize()} Metric"] + [f"DataLoader {i}" for i in range(len(results))]
+            max_length = max(len(max(unique_keys + cols, key=len)), 25)
+            row_format = "{:^{max_length}}" * len(cols)
+            # cap wide terminals to 120 columns, but go over if there are many table columns
+            term_size = max(min(120, os.get_terminal_size().columns), max_length * len(cols))
+
+            print("\u2500" * term_size)
+            print(row_format.format(*cols, max_length=max_length))
+            print("\u2500" * term_size)
+            for col, row in zip(unique_keys, rows):
+                print(row_format.format(col, *row, max_length=max_length))
+            print("\u2500" * term_size)
```

Comment:
Should we construct the full string and then do a single `print` call with it?

Reply:
I believe that is a cleaner solution, yes!

Reply:
Done